### PR TITLE
feat(authentication-service): added express middleware in sequence

### PR DESCRIPTION
**Pull request by sf-kansara**
RPMS-2771

## Description

1. Added check to execute express middlewares in sequence
2. Enable obf by env parameter in sandbox testing

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [X] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [X] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: sandbox/auth-multitenant-example/src/application.ts [hunk 1]
```diff
@@ -65,11 +67,11 @@ export class AuthMultitenantExampleApplication extends BootMixin(
     });
     // morgan middleware should work with or without swagger-stats
     this.bind(SFCoreBindings.EXPRESS_MIDDLEWARES).to([
-      morgan('MORGAN_LOG > :method :url :status')
+      morgan('MORGAN_LOG > :method :url :status'),
```
Comment (samarpan-b) [inline sandbox/auth-multitenant-example/src/application.ts:70]: why morgan ? we have logger

File: services/authentication-service/src/sequence.ts [hunk 2]
```diff
@@ -79,6 +82,15 @@ export class MySequence implements SequenceHandler {
         Remote Address = ${request.connection.remoteAddress}
         Remote Address (Proxy) = ${request.headers['x-forwarded-for']}`,
       );
+
+      if (this.expressMiddlewares?.length) {
+        const responseGenerated = await this.invokeMiddleware(
```
Comment (samarpan-b) [inline services/authentication-service/src/sequence.ts:87]: this line is important to be executed. Otherwise, it will throw weird errors for every API.

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: sandbox/auth-multitenant-example/src/application.ts (hunk 1 above)
Comment (sf-kansara) [reply sandbox/auth-multitenant-example/src/application.ts:70]: Morgan is to test feature of having response and non response generating express middlewares. Also adding other middlewares such as swagger stats should work with morgan.

File: services/authentication-service/src/sequence.ts (hunk 2 above)
Comment (samarpan-b) [reply services/authentication-service/src/sequence.ts:87]: `await this.invokeMiddleware` This is needed. Else, APIs will fail.

File: sandbox/auth-multitenant-example/src/application.ts (hunk 1 above)
Comment (samarpan-b) [reply sandbox/auth-multitenant-example/src/application.ts:70]: But why use morgan and introduce another logger, when we already have a logger.

Comment (sf-kansara) [reply sandbox/auth-multitenant-example/src/application.ts:70]: I'll remove it

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**